### PR TITLE
Use `curly_braces_in_flow_control_structures` for non-`flutter` packages

### PR DESCRIPTION
**Pull request by guidezpl**
Part of https://github.com/flutter/flutter/issues/104498

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_test/test/matchers_test.dart [hunk 1]
```diff
@@ -560,12 +563,15 @@ void main() {
...
         actions |= index;
-      for (final int index in SemanticsFlag.values.keys)
+      }
+      for (final int index in SemanticsFlag.values.keys) {
         // TODO(mdebbar): Remove this if after https://github.com/flutter/engine/pull/9894
-        if (SemanticsFlag.values[index] != SemanticsFlag.isMultiline)
+        if (SemanticsFlag.values[index] != SemanticsFlag.isMultiline) {
           flags |= index;
+      }
+        }
```
Comment (goderbauer) [inline packages/flutter_test/test/matchers_test.dart:574]: Some indentation got messed up here.

**Review by goderbauer — APPROVED** [review]
LGTM with one nit

**Comment by Hixie** [comment]
test-exempt: formatting change

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**